datumaro/plugins/yolo_detection_format/extractor.py

```diff
@@ -0,0 +1,331 @@
+from datumaro.components.media import Image
+from datumaro.util.image import (
+    DEFAULT_IMAGE_META_FILE_NAME,
+    IMAGE_EXTENSIONS,
+    ImageMeta,
+    load_image,
+    load_image_meta_file,
+)
+from datumaro.util.meta_file_util import has_meta_file, parse_meta_file
+from datumaro.util.os_util import split_path, extract_subset_name_from_parent, find_files
+
+from .format import YoloDetectionPath
+
+T = TypeVar("T")
+
+
+class YoloDetectionExtractor(SourceExtractor):
+    class Subset(Extractor):
+        def __init__(self, name: str, parent: YoloDetectionExtractor):
+            super().__init__()
+            self._name = name
+            self._parent = parent
+            self.items: Dict[str, Union[str, DatasetItem]] = OrderedDict()
+
+        def __iter__(self):
+            for item_id in self.items:
+                item = self._parent._get(item_id, self._name)
+                if item is not None:
+                    yield item
+
+        def __len__(self):
+            return len(self.items)
+
+        def categories(self):
+            return self._parent.categories()
+
+    def __init__(
+        self,
+        config_path: str,
+        image_info: Union[None, str, ImageMeta] = None,
+        urls: Optional[List[str]] = None,
+        *,
+        subset: Optional[str] = None
+    ) -> None:
+        META_FILE = YoloDetectionPath.META_FILE
+
+        if not osp.isdir(config_path):
+            raise DatasetImportError(f"{config_path} should be a directory.")
+        
+        if not urls:
+            raise DatasetImportError(
+                f"`urls` should be specified for {self.__class__.__name__}, "
+                f"if you want to import a dataset with using this {self.__class__.__name__} directly. "
+                "In most case, it happens by giving an incorrect format name to the import interface. "
+                "Please consider to import your dataset with this format name, 'yolo', "
+                "such as `Dataset.import_from(..., format='yolo')`."
+            )
+
+        super().__init__(subset=subset)
+
+        rootpath = self._get_rootpath(config_path)
+        self._path = rootpath
+
+        self._image_info = self.parse_image_info(rootpath, image_info)
+        self._urls = urls
+        self._img_files = self._load_img_files(rootpath)
+        self._ann_types = set()
+        
+        self._categories = {
+            AnnotationType.label: self._load_categories(
+                osp.join(self._path, META_FILE)
+            )
+        }
+
+        # config = YoloDetectionPath._parse_config(config_path)
+
+        # names_path = config.get("names")
+        # if not names_path:
+        #     raise InvalidAnnotationError(f"Failed to parse names file path from config")
+
+        # # The original format is like this:
+        # #
+        # # classes = 2
+        # # train  = data/train.txt
+        # # valid  = data/test.txt
+        # # names = data/obj.names
```

Comment:
Please remove irrelevant comments. Don't copy and past. Try to rethink each line of code.